mne/decoding/time_gen.py

```diff
@@ -914,18 +544,681 @@ def _predict(X, estimators):
+    y_true_ : list | np.ndarray, shape (n_samples,)
+        The categories used for scoring y_pred_.
+    scorer_ : object
+        scikit-learn Scorer instance.
+    scores_ : list of float, shape (n_times)
+        The scores (mean accuracy of self.predict(X) wrt. y.).
+
+    Notes
+    -----
+    The function is equivalent to the diagonal of GeneralizationAcrossTime()
+    """
+
+    def __init__(self, picks=None, cv=5, clf=None, times=None,
+                 predict_mode='cross-validation', scorer=None, n_jobs=1):
+        super(TimeDecoding, self).__init__(picks=picks, cv=cv, clf=None,
+                                           train_times=times,
+                                           test_times='diagonal',
+                                           predict_mode=predict_mode,
+                                           scorer=scorer, n_jobs=n_jobs)
+        self.times = self.train_times
+        delattr(self, 'test_times')
+        delattr(self, 'train_times')
+
+    def __repr__(self):
+        s = ''
+        if hasattr(self, "estimators_"):
+            s += "fitted, start : %0.3f (s), stop : %0.3f (s)" % (
+                self.train_times_['start'], self.train_times_['stop'])
+        else:
+            s += 'no fit'
+        if hasattr(self, 'y_pred_'):
+            s += (", predicted %d epochs" % len(self.y_pred_[0]))
+        else:
+            s += ", no prediction"
+        if hasattr(self, "estimators_") and hasattr(self, 'scores_'):
+            s += ',\n '
+        else:
+            s += ', '
+        if hasattr(self, 'scores_'):
+            s += "scored"
+            if callable(self.scorer_):
+                s += " (%s)" % (self.scorer_.__name__)
+        else:
+            s += "no score"
+
+        return "<TimeDecoding | %s>" % s
+
+    def fit(self, epochs, y=None):
+        """ Train a classifier on each specified time slice.
+
+        Note. This function sets the ``picks_``, ``ch_names``, ``cv_``,
+        ``y_train``, ``train_times_`` and ``estimators_`` attributes.
+
+        Parameters
+        ----------
+        epochs : instance of Epochs
+            The epochs.
+        y : list or np.ndarray of int, shape (n_samples,) or None, optional
+            To-be-fitted model values. If None, y = epochs.events[:, 2].
+
+        Returns
+        -------
+        self : object
+            Returns self.
+
+        Notes
+        ------
+        If X and y are not C-ordered and contiguous arrays of np.float64 and
+        X is not a scipy.sparse.csr_matrix, X and/or y may be copied.
+
+        If X is a dense array, then the other methods will not support sparse
+        matrices as input.
+        """
+        self.train_times = self.times
+        super(TimeDecoding, self).fit(epochs, y=y)
+        self.times_ = self.train_times_
+        # GAT compatibility unsqueeze testing times # XXX JRK: need cleanup
+        delattr(self, 'train_times_')
+        delattr(self, 'train_times')
```

Comment:
Mmmh, yeah that's what I meant by 'there's room for improvement'. It's safe, but it's not cleaned. The idea is to avoid having extra attributes available to the user. `TimeDecoding` only possesses a `times` attr, equivalent to `train_times`.